### DISCUSSION TITLE
How do I autoregress on regressors that aren't the target?

### DISCUSSION THREAD

**Discussion by tszumowski** [tinkoff-ai/etna#435] [Q&A]
I noticed the AutoRegressivePipeline autoregresses on the target variable to build out the multi-step forecast output. I have a variable that I lag alongside the target, because it is a useful regressor, but it is not something known at the time of forecast. 

For example, consider the case where trying to predict a store's product sales by week. The target is `sales`. However, we may have a regressor, `customer_visits` that indicates the number of times a customer walks in the store. If I am forecasting for the next 4 weeks, I do not know the values of `customer_visits` ahead of time. But I can forecast it alongside sales and use it to forecast `sales`. 

Is there a way to do this? If not, is there a suggested alternative?

**Comment by Mr-Geekman** [comment]
I can see two ways here:
1. You can use lagged version of the `customer_visits` with lag >= 4. In this case it will be available on the 4-th week.
2. You can try to build a separate model for prediction of `customer visits` on test data and then feed its predictions to model, that treats `customer_visits` as regressor. There is a problem with this approach: on train data we use true values of `customer_visits`, but on test data we use predicted values that are not precise but model will rely on it like it is precise.
